applications/client/src/main/java/com/paritytrading/philadelphia/client/command/WaitCommand.java

```diff
@@ -0,0 +1,56 @@
+package com.paritytrading.philadelphia.client.command;
+
+import com.paritytrading.philadelphia.client.TerminalClient;
+import com.paritytrading.philadelphia.client.message.Message;
+import com.paritytrading.philadelphia.client.message.Messages;
+import org.eclipse.collections.api.list.ImmutableList;
+
+import java.util.Scanner;
+
+class WaitCommand implements Command {
+    private static final long WAIT_TIME_MILLIS = 500;
+
+    @Override
+    public void execute(TerminalClient client, Scanner arguments) throws CommandException {
+        if (!arguments.hasNext())
+            throw new CommandException();
+
+        String msgType = arguments.next();
+
+        if (arguments.hasNext())
+            throw new CommandException();
+
+        while(true) {
+            try {
+                Thread.sleep(WAIT_TIME_MILLIS);
+
+                if(lastMsgType(client.getMessages()).equals(msgType))
+                    break;
+
+            } catch(InterruptedException e) {
+                break;
+            }
+        }
+    }
+
+    @Override
+    public String getName() {
+        return "wait";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Wait for a specific MsgType";
+    }
+
+    @Override
+    public String getUsage() {
+        return "wait <MsgType>";
+    }
+
+    private String lastMsgType(Messages messages) {
+        ImmutableList<Message> collect = messages.collect();
+
+        return collect.get(collect.size() - 1).getMsgType();
+    }
```

Comment:
This will fail if we haven't yet received any messages. What about:
```java
if (messages.isEmpty())
    return null;

return messages.getLast().getMsgType();
```

With the above, we could then simply switch the check in the loop:
```java
if (msgType.equals(getLastMsgType(client.getMessages())))
    break;
```

Finally, perhaps this could be called `getLastMsgType` to fit in with the existing methods. :wink: